test/integration/tcproute_test.go

```diff
@@ -432,6 +433,275 @@ func TestTCPRouteEssentials(t *testing.T) {
 	}, ingressWait, waitTick)
 }
 
+func TestTCPRouteReferencePolicy(t *testing.T) {
+	ns, cleanup := namespace(t)
+	defer cleanup()
+	t.Log("locking TCP port")
+	tcpMutex.Lock()
+	defer func() {
+		t.Log("unlocking TCP port")
+		tcpMutex.Unlock()
+	}()
+
+	other, err := clusters.GenerateNamespace(ctx, env.Cluster(), t.Name()+"other")
+	require.NoError(t, err)
+	defer func(t *testing.T) {
+		assert.NoError(t, clusters.CleanupGeneratedResources(ctx, env.Cluster(), t.Name()+"other"))
+	}(t)
+
+	// TODO consolidate into suite and use for all GW tests?
```

Comment:
There's actually a lot of refactoring we could do on the tests after the last few months of iterations and the advent of Gateway API code. I would be in favor of creating a `area/maintenance` issue here. We could start using that as a bullet point list of several testing refactors that we want to see done as we think of them. Perhaps we need to mark it as `nice-to-have` or `onboarding`.